sdks/python/apache_beam/testing/util_test.py

```diff
@@ -67,6 +73,26 @@ def test_assert_that_fails(self):
       with TestPipeline() as p:
         assert_that(p | Create([1, 10, 100]), equal_to([1, 2, 3]))
 
+  def test_assert_missing(self):
+    with self.assertRaisesRegex(BeamAssertException,
+                                r"missing elements \['c'\]"):
+      with TestPipeline() as p:
+        assert_that(p | Create(['a', 'b']), equal_to(['a', 'b', 'c']))
+
+  def test_assert_unexpected(self):
+    with self.assertRaisesRegex(BeamAssertException,
+                                r"unexpected elements \['c', 'd'\]"):
+      with TestPipeline() as p:
+        assert_that(p | Create(['a', 'b', 'c', 'd']), equal_to(['a', 'b']))
```

Comment:
PCollections don't guarantee the order of elements, so once PCollection materializes we may access its elements in the order of 'a', 'b', 'd', 'c', in which case the generated error message will be "unexpected elements ['d', 'c']". While also correct, this message will fail current assertion. Added a small change to accommodate that, and reversed the order in try clause to avoid a warning at runtime on Py3.